plugins/inputs/t128_tank/reader.go

```diff
@@ -0,0 +1,271 @@
+		accumulating:  nil,
+		lastReadIndex: indexFile.Last(),
+		count:         0,
+	}
+}
+
+func (lc *LineContext) RecordIndex(index int, shouldExpectEveryLine bool) {
+	if lc.count%recordIndexAfter == 0 {
+		lc.index.Save(index)
+	}
+
+	if shouldExpectEveryLine && !contains(specialIndices, lc.lastReadIndex) && index != (lc.lastReadIndex+1) {
+		log.Printf("Missed messages between indices %v and %v", lc.lastReadIndex, index)
+	}
+
+	lc.lastReadIndex = index
+	lc.count++
+}
+
+func (lc *LineContext) Save() {
+	lc.index.Save(lc.lastReadIndex)
+}
+
+func (lc *LineContext) Accumulated() string {
+	if lc.accumulating == nil {
+		return ""
+	}
+	return lc.accumulating.line
+}
+
+func (bf BoundaryFault) Error() string {
+	return "Boundary fault"
+}
+
+func YieldValidLines(topic int, indexFile *IndexFile, isMainThread bool, tankPath string, shouldExpectEveryLine bool) <-chan string {
+	result := make(chan string)
+
+	go func() {
+		readNext := indexFile.Last()
+
+		if !contains(specialIndices, readNext) {
+			readNext = readNext + 1
+		}
+
+		cliCommand := buildCLICommand(tankPath, topic, readNext)
+		cmd := exec.Command(cliCommand[0], cliCommand[1:]...)
+		cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+		stdout, err := cmd.StdoutPipe()
+		if err != nil {
+			fmt.Errorf("Error creating stdout pipe: " + err.Error())
+			close(result)
+			return
+		}
+
+		if err := cmd.Start(); err != nil {
+			fmt.Errorf("Error starting tank-cli: " + err.Error())
+			close(result)
+			return
+		}
+
+		defer cmd.Wait()
+
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			defer close(result)
+			lines := make(chan string)
+			go func() {
+				defer close(lines)
+				buf := make([]byte, 1024)
+				for {
+					n, err := stdout.Read(buf)
+					if n > 0 {
+						lines <- string(buf[:n])
+					}
+					if err != nil {
+						break
+					}
+				}
+			}()
+
+			for line := range lines {
+				tankLine, err := lineContextAdd(line, indexFile)
+				if err != nil {
+					if _, ok := err.(BoundaryFault); ok {
+						//
+					}
+					continue
+				}
+
+				if tankLine != nil {
+					result <- tankLine.line
+					lineContextRecordIndex(tankLine.index, shouldExpectEveryLine, indexFile)
+				}
+			}
+		}()
+
+		wg.Wait()
+	}()
+
+	return result
+
+}
+
+func lineContextAdd(line string, indexFile *IndexFile) (*TankLine, error) {
+	context := NewLineContext(indexFile)
+	tankLine, err := context.Add(line)
+	if err != nil {
+		return nil, err
+	}
+	return tankLine, nil
+}
+
+func lineContextRecordIndex(index int, shouldExpectEveryLine bool, indexFile *IndexFile) {
+	context := NewLineContext(indexFile)
+	context.RecordIndex(index, shouldExpectEveryLine)
+}
+
+func contains(specialIndices []interface{}, readNext int) bool {
+	for _, value := range specialIndices {
+		if value == readNext {
+			return true
+		}
+	}
+	return false
+}
+
+func buildCLICommand(tankPath string, topic int, readNext int) []string {
+	return []string{
+		tankPath,
```

Comment:
I thought we were going to try and use the go tank client?

Reply:
I tested it and it isn't quite what it needs to be yet. I'm going to work when I can to get it in shape, but it's not a great short term option.